src/fdb5/toc/TocHandler.cc

```diff
@@ -451,11 +451,11 @@ bool TocHandler::readNext(TocRecord& r, bool walkSubTocs, bool hideSubTocEntries
     }
 
     while (true) {
-
         if (subTocRead_) {
             len = subTocRead_->readNext(r, walkSubTocs, hideSubTocEntries, hideClearEntries, readMasked, data, length);
             if (len == 0) {
-                subTocRead_ = nullptr;
+                subTocRead_    = nullptr;
+                *parentTocPath = eckit::LocalPathName();
```

Comment:
WARNING! DANGER!

You have not checked if the parentTocPath pointer is null. If it is, this will segfault. Need to wrap this:

```
if (parentTocPath) {
    *parentTocPath = "";
```